app/src/main/java/com/kickstarter/viewmodels/PrivacyViewModel.kt

```diff
@@ -50,91 +51,97 @@ interface PrivacyViewModel {
 
             val updateSettingsNotification = this.userInput
                 .concatMap { this.updateSettings(it) }
 
             updateSettingsNotification
-                .compose(values())
-                .compose(bindToLifecycle())
+                .compose(valuesV2())
                 .subscribe { this.success(it) }
+                .addToDisposable(disposables)
 
             updateSettingsNotification
-                .compose(errors())
-                .compose(bindToLifecycle())
-                .subscribe(this.unableToSavePreferenceError)
+                .compose(errorsV2())
+                .filter { it.isNotNull() }
+                .subscribe { this.unableToSavePreferenceError.onNext(it!!) }
+                .addToDisposable(disposables)
 
             this.userInput
-                .compose(bindToLifecycle())
-                .subscribe(this.userOutput)
+                .filter { it.isNotNull() }
+                .subscribe { this.userOutput.onNext(it) }
+                .addToDisposable(disposables)
 
             this.userOutput
                 .window(2, 1)
-                .flatMap<List<User>> { it.toList() }
+                .flatMap<List<User>> { it.toList().toObservable() }
                 .map<User> { ListUtils.first(it) }
-                .compose<User>(Transformers.takeWhen<User, Throwable>(this.unableToSavePreferenceError))
-                .compose(bindToLifecycle())
-                .subscribe(this.userOutput)
+                .compose<User>(Transformers.takeWhenV2<User, Throwable>(this.unableToSavePreferenceError))
+                .subscribe { this.userOutput.onNext(it) }
+                .addToDisposable(disposables)
 
             this.optIntoFollowing
-                .compose<Boolean>(bindToLifecycle<Boolean>())
                 .filter { checked -> checked }
-                .subscribe { _ -> this.userInput.onNext(this.userOutput.value.toBuilder().social(true).build()) }
+                .filter { this.userOutput.value.isNotNull() }
+                .subscribe { _ -> this.userInput.onNext(this.userOutput.value!!.toBuilder().social(true).build()) }
+                .addToDisposable(disposables)
 
             this.optIntoFollowing
-                .compose<Boolean>(bindToLifecycle<Boolean>())
                 .filter { checked -> !checked }
-                .subscribe { _ -> this.showConfirmFollowingOptOutPrompt.onNext(null) }
+                .subscribe { _ -> this.showConfirmFollowingOptOutPrompt.onNext(Unit) }
+                .addToDisposable(disposables)
 
             this.optOutOfFollowing
-                .compose<Boolean>(bindToLifecycle<Boolean>())
                 .filter { optOut -> optOut }
-                .subscribe { _ -> this.userInput.onNext(this.userOutput.value.toBuilder().social(false).build()) }
+                .filter { this.userOutput.value.isNotNull() }
+                .subscribe { _ -> this.userInput.onNext(this.userOutput.value!!.toBuilder().social(false).build()) }
```

Comment:
same as above, we have historically avoided using `!!`, tho im open to revisiting this